pkg/arvo/sys/vane/clay.hoon

```diff
@@ -931,6 +974,29 @@
       =.  p.pin  [%face face.i.raw p.pin]
       $(sut (slop pin sut), raw t.raw)
     ::
+    ++  run-raz
+      |=  [sut=vase raz=(list [face=term =spec =path])]
+      ^-  [vase state]
+      ?~  raz  [sut nub]
+      =^  res=(map @ta vase)  nub
+        (build-directory path.i.raz)
+      =;  pin=vase
+        =.  p.pin  [%face face.i.raz p.pin]
+        $(sut (slop pin sut), raz t.raz)
+      ::
+      =/  =type  (~(play ut p.sut) [%kttr spec.i.raz])
+      ::  ensure results nest in the specified type,
+      ::  and produce a homogenous map containing that type.
+      ::
+      :-  %-  ~(play ut p.sut)
+          [%kttr %make [%wing ~[%map]] ~[[%base %atom %ta] spec.i.raz]]
+      |-
+      ?~  res  ~
+      ~|  [%nest-fail path.i.raz p.n.res]
```

Comment:
I think this will print a whole list of failures, instead of just the single nest failure.  Maybe put the nest check inside the head of the `:-` expression so that further recursive cases aren't subexpressions of the `~|`.